logplex_formatter.go

```diff
@@ -21,16 +22,26 @@ const (
 type LogplexBatchFormatter struct {
 	headers       http.Header
 	stringURL     string
+	user, pass    string
 	msgCount      int
 	contentLength int64
 	io.Reader
 }
 
 // NewLogplexBatchFormatter returns a new LogplexBatchFormatter wrapping the provided batch
 func NewLogplexBatchFormatter(b Batch, eData []errData, config *Config) HTTPFormatter {
+	u, err := url.Parse(config.LogsURL)
+	if err != nil {
+		panic("impossible")
```

Comment:
I checked and we definitely ensure `config.LogsURL` is valid at the edges. I'm open to ideas, but seems like it's unnecessary to make the return type here be `(HTTPFormatter, error)` just for this.

Reply:
Can this be done the first time Request is called and stringURL is used? Then an error could be returned.

Reply:
that's a great idea for returning the error and also nicely handles the case (I believe) in the kinesis formatter.